### PR TITLE
HS-1162 - add severity filter, label component for severities

### PR DESCRIPTION
**Pull request by anyarybalova**
## Description
- add the severity filters box
- label severity component
- the colours  based on https://www.figma.com/file/7BWMGTBLpSp44TdNIPrd3p/Status-Tag?node-id=101%3A34, for now I added as custom, Amanda tries to check if they can be added ASAP to feather

## Jira link(s)
- https://issues.opennms.org/browse/HS-1162

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: ui/src/components/Alerts/AlertsSeverityLabel.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+<style lang="scss" scoped>
+@use '@featherds/styles/themes/variables';
+@use '@featherds/styles/mixins/typography';
+.severity-status {
+  padding: 2px 8px;
+  font-size: 11px;
+  border-radius: 5px;
+  font-weight: 600;
+  width: fit-content;
```
Comment (chinh-cle) [inline ui/src/components/Alerts/AlertsSeverityLabel.vue:24]: not sure if this does anything.

**Review by chinh-cle — CHANGES_REQUESTED** [review]
I suggest that you run yarn build to ensure it builds without any ts errors.

File: ui/src/components/Alerts/AlertsSeverityCard.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+.icon-anim-enter-active {
+  transition: all 0.3s ease-out;
+}
+
+.icon-anim-leave-active {
+  transition: all 0.8s cubic-bezier(1, 0.5, 0.8, 1);
+}
+
+.icon-anim-enter-from,
+.icon-anim-leave-to {
```
Comment (mikewrosey) [inline ui/src/components/Alerts/AlertsSeverityCard.vue:90]: Will we reuse there? just wondering if we want to move these to a separate styles folder file, like we do with `_transitionFade.scss` ?

File: ui/src/store/Views/alertsStore.ts [hunk 3]
```diff
@@ -24,6 +26,11 @@ export const useAlertsStore = defineStore('alertsStore', () => {
     } else {
       severitiesSelected.value.push(selected)
     }
+    if (!severitiesSelected.value.length) {
+      alertList.value = allAlertsList.value
+    } else {
+      alertList.value = allAlertsList.value.filter((a) => severitiesSelected.value.includes(a.severity))
+    }
```
[the remark this answers is not in the document]
Comment (anyarybalova) [reply ui/src/store/Views/alertsStore.ts:33]: its temporary, just to see the filters are working

File: ui/src/components/Alerts/AlertsSeverityCard.vue (hunk 2 above)
Comment (anyarybalova) [reply ui/src/components/Alerts/AlertsSeverityCard.vue:90]: when we are going to reuse, we can move it then

File: ui/src/components/Alerts/AlertsSeverityCard.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<template>
+  <div
+    class="card border pointer"
+    :class="{ selected: !isAdd }"
+    @click="addSeverityFilter"
+  >
+    <div class="row">
+      <div><AlertsSeverityLabel :severity="severity" /></div>
```
[the remark this answers is not in the document]
Comment (anyarybalova) [reply ui/src/components/Alerts/AlertsSeverityCard.vue:8]: hmm I don't have this error

File: ui/src/store/Views/alertsStore.ts [hunk 5]
```diff
@@ -24,6 +26,11 @@ export const useAlertsStore = defineStore('alertsStore', () => {
     } else {
       severitiesSelected.value.push(selected)
     }
+    if (!severitiesSelected.value.length) {
```
[the remark this answers is not in the document]
Comment (anyarybalova) [reply ui/src/store/Views/alertsStore.ts:29]: yeah, we don't have type alert

**Review by chinh-cle — APPROVED** [review]

**Pull request merged**